File: backend/routes/api/v1/role.go
package v1

import (
	"datax-admin/controllers"
	"github.com/gin-gonic/gin"
)

// RegisterRoleRoutes 注册角色管理相关路由
func RegisterRoleRoutes(authenticated *gin.RouterGroup) {
	roleController := controllers.NewRoleController()
	
	// 角色管理相关
	authenticated.POST("/roles", roleController.CreateRole)
	authenticated.PUT("/roles/:id", roleController.UpdateRole)
	authenticated.DELETE("/roles/:id", roleController.DeleteRole)
	authenticated.GET("/roles", roleController.GetRoleList)
	authenticated.GET("/roles/:id/permissions", roleController.GetRolePermissions)
	authenticated.PUT("/roles/:id/permissions", roleController.UpdateRolePermissions)

	// 用户角色管理
	authenticated.GET("/users/:id/roles", roleController.GetUserRoles)
	authenticated.PUT("/users/:id/roles", roleController.UpdateUserRoles)
}

File: web/src/api/role.ts
import request from '@/utils/request';
import type {
  CreateRoleParams,
  UpdateRoleParams,
  RoleInfo,
  RoleListParams,
  RoleListResult,
} from '@/types/role';

export function createRole(data: CreateRoleParams) {
  return request.post('/roles', data);
}

export function updateRole(id: number, data: UpdateRoleParams) {
  return request.put(`/roles/${id}`, data);
}

export function deleteRole(id: number) {
  return request.delete(`/roles/${id}`);
}

export function getRoleList(params: RoleListParams) {
  return request.get<RoleListResult>('/roles', { params });
}

export function getRolePermissions(roleId: number) {
  return request.get(`/roles/${roleId}/permissions`);
}

export function updateRolePermissions(roleId: number, permissionIds: number[]) {
  return request.put(`/roles/${roleId}/permissions`, { permission_ids: permissionIds });
}

export function getUserRoles(userId: number) {
  return request.get(`/users/${userId}/roles`);
}

export function updateUserRoles(userId: number, roleIds: number[]) {
  return request.put(`/users/${userId}/roles`, { role_ids: roleIds });
}
